src/deployer.h

```diff
@@ -33,41 +33,23 @@
 
 #include <QObject>
 #include <QStringList>
-
-#include <Transaction>
+#include <QDBusReply>
+#include <QDBusInterface>
 
 class Deployer : public QObject {
     Q_OBJECT
 
 public:
-    Deployer(QStringList rpms, bool verbose=false);
+    Deployer(QStringList rpms);
     ~Deployer();
 
 public slots:
-    void run();
-    void onChanged();
-    void onItemProgress(const QString &itemID,
-            PackageKit::Transaction::Status status,
-            uint percentage);
-    void onFinished(PackageKit::Transaction::Exit status,
-            uint runtime);
-    void onErrorCode(PackageKit::Transaction::Error error,
-                     const QString &details);
-    void onPackage(PackageKit::Transaction::Info info,
-                   const QString &packageID,
-                   const QString &summary);
+    QDBusReply<void> run();
```

Comment:
Minor, but wouldn't need to be a slot as it's called only directly once.

QDBusReply in type is a bit leaky, but can live with that with this small project.